HCDevice.py

```diff
@@ -252,6 +265,47 @@ def get(self, resource, version=1, action="GET", data=None):
             print(self.name, "Failed to send", e, msg, traceback.format_exc())
         self.tx_msg_id += 1
 
+    def reconnect(self):
+        self.ws.reconnect()
+        # Receive initialization message /ei/initialValues
+        # Automatically responds in the handle_message function
+        self.recv()
+
+        # ask the device which services it supports
+        # registered devices gets pushed down too hence the loop
+        self.get("/ci/services")
+        while True:
+            self.recv()
+            if self.services_initialized:
+                break
+
+        # We override the version based on the registered services received above
+
+        # the clothes washer wants this, the token doesn't matter,
+        # although they do not handle padding characters
+        # they send a response, not sure how to interpet it
+        self.token = base64url_encode(get_random_bytes(32)).decode("UTF-8")
+        self.token = re.sub(r"=", "", self.token)
+        self.get("/ci/authentication", version=2, data={"nonce": self.token})
+
+        self.get("/ci/info")  # clothes washer
+        self.get("/iz/info")  # dish washer
```

Comment:
We should check that `iz` `ci` etc are in services. We shouldn't send them due to checks further in the code but clearer to check here too.